"Updated offensive behaviors, and…

ErfanFathii opened the pull request:
… modified KickDecisionMaker to use BhvStarterKickPlanner."

ErfanFathii committed:
"Updated BhvStarterKickPlanner to import and use other behaviors, and modified KickDecisionMaker to use BhvStarterKickPlanner."

naderzare commented on src/behaviors/bhv_starter_pass.py:
```diff
@@ -0,0 +1,43 @@
+from service_pb2 import *
+
+
+class Pass(IBehavior):
```
class name should be BhvStarterPass

naderzare commented on src/behaviors/bhv_starter_dribble.py:
```diff
@@ -0,0 +1,33 @@
+from service_pb2 import *
+
+
+class Dribble(IBehavior):
```
BhvStarterDribble

naderzare commented on src/behaviors/bhv_starter_clearball.py:
```diff
@@ -0,0 +1,39 @@
+from pyrusgeom import vector_2d
+from service_pb2 import *
+
+class ClearBall(IBehavior):
```
Bhv...

naderzare commented on src/decision_makers/kick_decision_maker.py:
```diff
@@ -10,10 +10,11 @@
         agent.logger.debug("--- WithBallDecisionMaker ---")
         from src.sample_player_agent import SamplePlayerAgent  # Local import to avoid circular import
         assert isinstance(agent, SamplePlayerAgent)
+        #self.bhv_kick_planner.execute(agent)
```
we do not need this line

naderzare commented on src/decision_makers/kick_decision_maker.py:
```diff
@@ -10,10 +10,11 @@
     def __init__(self):
         self.bhv_kick_planner = BhvKickPlanner()
-        # self.bhv_kick_planner = BhvStarterKickPlanner()
+        self.bhv_kick_planner = BhvStarterKickPlanner()
```
for default we should not use Starter classes, but we can have a variable in sample player agent, for example starter=true/false and we can choose planner or starter planner based on that variable

SoroushMazloum committed:
fix class names and kick_decision_maker

SoroushMazloum reviewed (commented):
I fixed these issues in my new commit

SoroushMazloum committed:
add starter tackle

SoroushMazloum committed:
edit starter clearball, dribble, kick_planner, pass, shoot. add some setplay files: go_to_placed_ball, intention_wait_after_setplay, prepare_setplay_kick, setplay_freekick. add starter_strategy

SoroushMazloum committed:
add more setplay files

SoroushMazloum committed:
add penalty cycle to .proto, starter_setplay is done

SoroushMazloum committed:
Revert "add penalty cycle to .proto, starter_setplay is done"

This reverts commit 228f38900ca7ecd7c3a73d619285d5ccb495a86b.
(Erfan should add this part)

ErfanFathii committed:
`Added new field to PenaltyKickState message and updated import in set_play_decision_maker.py. Added two new methods to tools.py: BallInertiaFinalPoint and OpponentGoalie. Also added some setplay files`

ErfanFathii committed:
feat: refine `penalty kick` logic for kicker and goalie

- Designed penalty kick behavior in `BhvStarterPenalty`:
  - Added kicker setup, readiness, and execution methods.
  - Implemented goalie actions: setup, waiting, and responses.
  - Enhanced decision-making for `dribbling` and `shooting`.
  - Improved goalie positioning based on ball dynamics.
  - Included support for various penalty game modes.
